suggestionBox/suggestionBox-war/src/main/java/org/silverpeas/components/suggestionbox/common/SuggestionBoxWebManager.java

```diff
@@ -253,7 +253,7 @@ public SuggestionEntity publishSuggestion(SuggestionBox suggestionBox, Suggestio
     if (suggestion.isDefined() && (suggestion.getValidation().isInDraft() || suggestion.
         getValidation().isRefused())) {
       checkAdminAccessOrUserIsCreator(fromUser, suggestion);
-      suggestion.setLastUpdater(fromUser);
+      suggestion.updatedBy(fromUser);
       Suggestion actual = suggestionBox.getSuggestions().publish(suggestion);
       UserPreferences userPreferences = fromUser.getUserPreferences();
       switch (actual.getValidation().getStatus()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Complete cases by adding the missing enum constants or add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Switch statement found in org.silverpeas.components.suggestionbox.common.SuggestionBoxWebManager.publishSuggestion(SuggestionBox, Suggestion, User) where default case is missing [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=findbugs%3ASF_SWITCH_NO_DEFAULT)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1301)